js/repl/src/repl/Repl.js

```diff
@@ -0,0 +1,327 @@
+      persistedState
+    );
+
+    // Load any plug-ins enabled by query params
+    this._checkForUnloadedPlugins();
+  }
+
+  render() {
+    const {
+      code,
+      compiled,
+      compileError,
+      evaluate,
+      evalError,
+      lineWrap,
+      plugins,
+      presets
+    } = this.state;
+
+    const options = {
+      lineWrapping: lineWrap
+    };
+
+    return (
+      <div className={styles.repl}>
+        <ReplOptions
+          className={styles.optionsColumn}
+          evaluate={evaluate}
+          lineWrap={lineWrap}
+          pluginState={plugins}
+          presetState={presets}
+          toggleSetting={this._toggleSetting}
+        />
+
+        <div className={styles.panels}>
+          <CodeMirrorPanel
+            className={styles.codeMirrorPanel}
+            code={code}
+            error={compileError}
+            onChange={this._updateCode}
+            options={options}
+          />
+          <CodeMirrorPanel
+            className={styles.codeMirrorPanel}
+            code={compiled}
+            error={evalError}
+            options={options}
+          />
+        </div>
+      </div>
+    );
+  }
+
+  _checkForUnloadedPlugins() {
+    const { plugins } = this.state;
+
+    // Assume all default presets are baked into babel-standalone
+    // We really only need to worry about plugins
+    for (const key in plugins) {
+      const plugin = plugins[key];
+
+      if (plugin.isEnabled && !plugin.isLoaded && !plugin.isLoading) {
+        this._numLoadingPlugins++;
+
+        loadPlugin(plugin, success => {
+          this._numLoadingPlugins--;
+
+          if (!success) {
+            this.setState(state => ({
+              plugins
+            }));
+          }
+
+          if (this._numLoadingPlugins === 0) {
+            const { code } = this.state;
+
+            this._updateCode(code);
+          }
+        });
+      }
+    }
+  }
+
+  _compile = (code: string, state: State) => {
+    const { evaluate, plugins } = state;
+
+    return compile(code, {
+      evaluate: evaluate,
+      presets: this._presetsToArray(state),
+      prettify: plugins.prettier.isEnabled
+    });
+  };
+
+  _loadState(): PersistedState {
+    const storageState = StorageService.get('replState');
+    const queryState = UriUtils.parseQuery();
+    const merged = {
+      ...storageState,
+      ...queryState
+    };
+
+    return {
+      babili: merged.babili === true,
+      browsers: merged.browsers || '',
+      builtIns: merged.builtIns === true,
+      code: merged.code || '',
+      debug: merged.debug === true,
+      evaluate: merged.evaluate === true,
+      lineWrap: merged.lineWrap != null ? merged.lineWrap : true,
+      presets: merged.presets || '',
+      prettier: merged.prettier === true,
+      targets: merged.targets || ''
+    };
+  }
+
+  _presetsToArray(state: State = this.state): Array<string> {
+    const { presets } = state;
+
+    return Object.keys(presets)
```

Comment:
If this is being used as a map, could we use a `Map` rather than an object? Given the latest React beta requires `Map` to be polyfilled, we can likely rely on it existing?